lib/ansible/modules/cloud/amazon/ecs_tag.py

```diff
@@ -0,0 +1,238 @@
+version_added: '2.10'
+author:
+  - Lester Wade (@lwade)
+  - Paul Arthur (@flowerysong)
+  - Michael Pechner (@mpechner)
+requirements: [ boto3, botocore ]
+options:
+  cluster_name:
+    description:
+      - The name of the cluster who's resources we are tagging
+    required: true
+    type: str
+  resource:
+    description:
+      - The ecs resource name. Can be none only if resource_type is cluster.
+    required: false
+    type: str
+  resource_type:
+    description:
+      - The type of resource
+    default: cluster
+    choices: ['cluster', 'task', 'service', 'task_definition', 'container']
+    type: str
+  state:
+    description:
+      - Whether the tags should be present or absent on the resource. Use list to interrogate the tags of a ecs resource.
+    default: present
+    choices: ['present', 'absent', 'list']
+    type: str
+  tags:
+    description:
+      - A dictionary of tags to add or remove from the resource.
+      - If the value provided for a tag is null and C(state) is I(absent), the tag will be removed regardless of its current value.
+    type: dict
+  purge_tags:
+    description:
+      - Whether unspecified tags should be removed from the resource.
+      - "Note that when combined with C(state: absent), specified tags with non-matching values are not purged."
+    type: bool
+    default: no
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = r'''
+- name: Ensure tags are present on a resource
+  ecs_tag:
+    cluster_name: mycluster
+    resource_type: cluster
+    state: present
+    tags:
+      Name: ubervol
+      env: prod
+
+- name: Retrieve all tags on a cluster
+  ecs_tag:
+    cluster_name: mycluster
+    resource: http_task
+    resource_type: task
+    state: list
+
+- name: Remove the Env tag
+  ecs_tag:
+    cluster_name: mycluster
+    resource_type: cluster
+    tags:
+      Env:
+    state: absent
+
+- name: Remove the Env tag if it's currently 'development'
+  ecs_tag:
+    cluster_name: mycluster
+    resource_type: cluster
+    tags:
+      Env: development
+    state: absent
+
+- name: Remove all tags except for Name from a cluster
+  ecs_tag:
+    cluster_name: mycluster
+    resource_type: cluster:27
+    tags:
+        Name: foo
+    state: absent
+    purge_tags: true
+'''
+
+RETURN = r'''
+tags:
+  description: A dict containing the tags on the resource
+  returned: always
+  type: dict
+added_tags:
+  description: A dict of tags that were added to the resource
+  returned: If tags were added
+  type: dict
+removed_tags:
+  description: A dict of tags that were removed from the resource
+  returned: If tags were removed
+  type: dict
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_tag_list_to_ansible_dict, ansible_dict_to_boto3_tag_list, compare_aws_tags
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except Exception:
```

Comment:
```suggestion
except ImportError:
```

It's generally considered a 'bad' thing to catch 'Exception'.  This is normally done by catching ImportError...